filters/flowid/generator-ulid.go

```diff
@@ -0,0 +1,40 @@
+package flowid
+
+import (
+	"github.com/oklog/ulid"
+	"io"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type ulidGenerator struct {
+	sync.Mutex
+	r io.Reader
+}
+
+func newULIDGenerator() flowIDGenerator {
+	return newULIDGeneratorWithEntropyProvider(rand.New(rand.NewSource(time.Now().UTC().UnixNano())))
+}
+
+func newULIDGeneratorWithEntropyProvider(r io.Reader) flowIDGenerator {
+	return &ulidGenerator{r: r}
+}
+
+func (g *ulidGenerator) Generate() (string, error) {
+	g.Lock()
```

Comment:
this is somewhat fishy to me. I see https://github.com/oklog/ulid#usage, but I wonder what is the cost of calling rand.New() + reading those few bytes from it on every request. I see the chance that it is better than both doing the synchronization here or using a sync.Pool. Let's do some parallel benchmark on it some time.